app/scripts/controllers/secret.js

```diff
@@ -45,13 +65,21 @@ angular.module('openshiftConsole')
 
         DataService.get("secrets", $scope.secretName, context, { errorNotification: false }).then(
           function(secret) {
+            secretResolved(secret);
+            watches.push(DataService.watchObject("secrets", $scope.secretName, context, secretResolved));
             $scope.secret = secret;
-            $scope.decodedSecretData = SecretsService.decodeSecretData($scope.secret.data);
-            $scope.loaded = true;
           },
           function(e) {
             $scope.loaded = true;
-            $scope.error = e;
+            $scope.alerts["load"] = {
+              type: "error",
+              message: "The secret details could not be loaded.",
+              details: $filter('getErrorDetails')(e)
+            };
           });
+
+        $scope.$on('$destroy', function(){
+          DataService.unwatchAll(watches);
+        });
```

Comment:
you should pass a final opts parameter to the get with `{errorNotification: false}` so that you don't get both a toast notification and the inline error

Reply:
Looks like it's already set in the call to `DataService.get`

Reply:
@jwforres not really sure what you mean

Reply:
nevermind @jhadvig i see it being passed above now, ignore this